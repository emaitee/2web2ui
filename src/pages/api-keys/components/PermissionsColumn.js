import React from 'react';
import { Tooltip } from '@sparkpost/matchbox';

import styles from './Permissions.module.scss';

const PermissionsColumn = ({ grants }) => {
  if (!grants.length) {
    return <span>None</span>;
  }

  const tooltipContent = (
    <span className={styles.Permissions}>
      {grants.join(', ')}
    </span>
  );

  return (
    <Tooltip content={tooltipContent} right horizontalOffset="-24px">
      {grants.length}
    </Tooltip>
  );
};

export default PermissionsColumn;
