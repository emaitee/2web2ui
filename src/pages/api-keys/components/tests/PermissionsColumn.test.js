import { shallow } from 'enzyme';
import React from 'react';

import PermissionsColumn from '../PermissionsColumn';

const props = {
  label: 'Test Key 1',
  short_key: 'ab01',
  grants: ['account/view', 'metrics/view', 'suppresions/view']
};

let wrapper;

beforeEach(() => {
  wrapper = shallow(<PermissionsColumn {...props} />);
});

it('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with no grants', () => {
  wrapper.setProps({ grants: []});
  expect(wrapper).toMatchSnapshot();
});
