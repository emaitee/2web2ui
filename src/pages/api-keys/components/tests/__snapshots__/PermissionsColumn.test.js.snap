// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<t
  bottom={true}
  content={
    <span>
      account/view, metrics/view, suppresions/view
    </span>
  }
  horizontalOffset="-24px"
  left={true}
  right={true}
>
  3
</t>
`;

exports[`renders correctly with no grants 1`] = `
<span>
  None
</span>
`;
